python/chemiscope/jupyter.py

```diff
@@ -246,7 +246,11 @@ def show(
             # also adds an index property to have something to show in the info panel
             properties["index"] = {
                 "target": "structure",
-                "values": list(range(len(frames))),
+                "values": (
+                    list(range(len(frames)))
+                    if hasattr(frames, "__len__")
+                    else list(range(len(frames.trajectory)))
```

Comment:
I don't love adding special cases everywhere in the code =/ Could it make sense to change the API such that the user would call `chemiscope.show(u.trajectory)` instead of `chemiscope.show(u)`? It does not seems that the code needs the other data from a `Universe`.

Reply:
No, I tried to pass the `u.trajectory`, but it does not contain the atom name information, which seems to be only stored in the `Universe`.

Reply:
Just had some more look into it. The information is also stored in `u.atoms`, and it has all the information that we need, but it follows a lazy loading way. Even though with the expression `[u.atoms.copy() for _ in u.trajectory]`, every `AtomGroup` object in the list still points to the first frame of the trajectory. If we can solve this problem, we can pass `list[AtomGroup]` instead of passing a `Universe`.